### ISSUE TITLE
Support for wildcards in command line

### ISSUE THREAD

**Issue by prabirshrestha** [microsoft/TypeScript#911]
It would be great if we could have wildcard support for files.

``` bash
tsc -w -m amd -t es5 **/*.ts
```

Currently we need to use unix commands.

``` unix
find . -name "*.ts" | xargs tsc -w -m amd -t es5
```

There are also cases when we would like to have wildcards for multiple different folders

``` bash
tsc -w -m amd -t es5 _generated/**/*.ts Typescript/**/*.ts
```

**Comment by prabirshrestha** [comment]
Also seems like `tsconfig.json` doesn't support wildcards.

**Comment by danquirk** [comment]
Correct, tsconfig doesn't currently support wildcards or globs. There are bugs already tracking the suggestion for that.

**Comment by mhegazy** [comment]
related to glob support in tsconfig: https://github.com/Microsoft/TypeScript/issues/1927

**Comment by NaridaL** [comment]
How would one go about creating a test for this? I've found https://github.com/Microsoft/TypeScript/blob/master/tests/cases/unittests/matchFiles.ts, would something similar be applicable or is that too low-level?

**Comment by mhegazy** [comment]
I do not think this file is used any more. looks like it was left there as a result of a bad merge. we should remove it.

The unit tests are under src/harness. the one closest is https://github.com/Microsoft/TypeScript/blob/master/src/harness/unittests/commandLineParsing.ts

that said, i think the importance of this issue is fairly low now. users have been using tsconfig.json, and we have not heard about this issue in over 2.5 years. so i would not spend much time on it. i think you can get better bang for your buck looking at other more recent issues.

**Comment by R3dXIII** [comment]
When I run tsc command I get file not found when using wildcards:
C:\Websites\hello-world>tsc *.ts && node main.js
    error TS6053: File '*.ts' not found.
    Found 1 error.

**Comment by hugovidafe** [comment]
Entering 2023, and we're exactly like 2014, amazing

**Comment by cawoodm** [comment]
I find it hard to believe that `tsc` doesn't support wildcards. But it doesn't. Why do I have to tell it which individual files to compile?


### Works Fine with individual files

```
tsc src/tmp.js --noEmit --checkJs
src/tmp.js:15:14 - error TS2345: Argument of type 'string' is not assignable to parameter of type 'number'.

15   return add('a', 2);
                ~~~
```

### Will not compile all files

```
tsc src/*.js --noEmit --checkJs
error TS6053: File 'src/*.js' not found.
  The file is in the program because:
    Root file specified for compilation


Found 1 error.
```

**Comment by RyanCavanaugh** [comment]
Shell expansion or tsconfig `include` seem like sufficient solutions here. Not really enough interest to warrant changing this.
